src/libraries/System.Private.CoreLib/src/System/IntPtr.cs

```diff
@@ -210,6 +211,10 @@ public int CompareTo(nint value)
         public bool TryFormat(Span<char> destination, out int charsWritten, [StringSyntax(StringSyntaxAttribute.NumericFormat)] ReadOnlySpan<char> format = default, IFormatProvider? provider = null) =>
             ((nint_t)_value).TryFormat(destination, out charsWritten, format, provider);
 
+        /// <inheritdoc/>
+        bool IUtf8SpanFormattable.TryFormat(Span<byte> utf8Destination, out int bytesWritten, ReadOnlySpan<char> format, IFormatProvider? provider) =>
+            ((IUtf8SpanFormattable)(nint_t)_value).TryFormat(utf8Destination, out bytesWritten, format, provider);
```

Comment:
Why would it box? Won't it be a constrained call?

Reply:
C# doesn't emit a constrained call here, because its just a standard instance method. It emits a `ldfld`, `box`, `ldarg...`, `callvirt instance...`

This requires the JIT to devirtualize things (and it may, but not all runtimes do this as well as others).

Reply:
This might be a place where C# could emit a constrained call to help the JIT, but I guess that'd require some work on Roslyn and confirmation it doesn't violate any language rules....

Reply:
I don't think it's worth complicating the implementation for. To my knowledge, the JIT has avoided the box here for a long time. If we for other reasons choose to make the methods public, that's fine, but I don't want to introduce backdoors to handle other code generators that may not handle things as well. The JIT does loads of optimizations we rely on without fallbacks.

Reply:
> This might be a place where C# could emit a constrained call to help the JIT, but I guess that'd require some work on Roslyn and confirmation it doesn't violate any language rules....

I'll open an issue if there isn't one already.